### DISCUSSION TITLE
Render anchor as next/link becomes a Fragment instead of Link when rendered in React component

### DISCUSSION THREAD

**Discussion by stahlmanDesign** [markdoc/markdoc#270] [Bug report] [answered]
### What happened?

TL;DR
config.nodes.link.render = Link results in Fragment, not Link, when parsing inside NextJS component

---
A raw markdown string parsed with Markdoc in the component becomes an anchor tag, as expected.

But when using NextJS, we don't want anchor tags because they reload the page. NextJS uses its custom Link instead.

But when we use Markdoc's config.nodes.link option to render link as a Link, it results in a Fragment instead of a Link, as shown in example

### To reproduce

Reproduced in [Codesandbox](https://codesandbox.io/p/sandbox/zealous-easley-x4ee63) 

### Version

^0.1.13

### Additional context

When using
```
const withMarkdoc = require('@markdoc/next.js')
const nextConfig = {...}
module.exports = withMarkdoc()(nextConfig)
```

to import a page in NextJS directly, this problem is not present. It is only present when parsing inside a React component using markdoc.transform and markdoc.renders

**Comment by mfix-stripe** [comment, marked as answer]
@stahlmanDesign sorry about this, looks like you ran into a few issues at once. I have forked your sandbox and [fixed all the issues here](https://codesandbox.io/p/sandbox/shy-voice-r1ooiw?file=%2Fpages%2Findex.tsx&selection=%5B%7B%22endColumn%22%3A18%2C%22endLineNumber%22%3A35%2C%22startColumn%22%3A18%2C%22startLineNumber%22%3A35%7D%5D&workspace=%257B%2522activeFileId%2522%253A%2522cl8skx9ay000hlpge4cua49bz%2522%252C%2522openFiles%2522%253A%255B%2522%252Fpages%252Findex.tsx%2522%255D%252C%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522gitSidebarPanel%2522%253A%2522COMMIT%2522%252C%2522sidekickItems%2522%253A%255B%257B%2522type%2522%253A%2522PREVIEW%2522%252C%2522taskId%2522%253A%2522dev%2522%252C%2522port%2522%253A3000%252C%2522key%2522%253A%2522cl9n4hc6800433b6h2r34jjgc%2522%252C%2522isMinimized%2522%253Afalse%257D%255D%257D).

1. Content in code elemenst (between single backticks) are not parsed — they are treated as raw strings. So when you had `[test](/test)` between backticks, it wasn't even being treated as a link. 
2. You need to pass `{ components: {...} }` to the 3rd parameter of `Markdoc.renderers.react`. I would recommend reading through this guide for end-to-end instructions on using Markdoc with React: https://markdoc.dev/docs/examples/react
3. Instead of `render: Link` you need `render: "Link"`. The mapping between the `"Link"` and the `Link` component occurs in the `components` map (from step 2). The reason this is required is because all content in the AST must be serializable, so we pass component names (e.g. `"Link"`) instead of the components themselves, and map them back together on the frontend. 

Hope that helps!

**Reply by stahlmanDesign** [reply]
Thank you, this is great and very helpful
